backend/main/src/Group/Domain/Entities/Container.ts

```diff
@@ -91,28 +95,37 @@ export class Container extends AggregateRoot<string, IContainerProps> {
 	}
 
 	/**
-	 * Update food object in the container.
+	 * Update food object in the container. But the createdAt is not updated.
 	 * If the food object doesn't exist in the container, return error.
 	 * @param food
 	 */
-	public updateFood(food: Food): Result<Container, ContainerDomainError> {
-		if (!this.props.foods.find((f) => f.id.equal(food.id))) {
+	public updateFood(
+		foodId: FoodId,
+		props: IFoodPropsWithoutCreatedAt,
+	): Result<Container, ContainerDomainError> {
+		if (!this.props.foods.find((f) => f.id.equal(foodId))) {
 			return Err(
 				new ContainerDomainError(
 					"The food object doesn't exist in the container",
 				),
 			);
 		}
-		// replace the food object
-		const foods = this.props.foods.map((f) => {
-			if (f.id.equal(food.id)) {
-				return food;
+		const newFoods: Food[] = [];
+		for (const food of this.props.foods) {
+			if (food.id.equal(food.id)) {
```

Comment:
This condition is always true. See also the comment on the update food use case test. Before fixing this condition, implement that test.
This code also ignores the situation when this condition is false.